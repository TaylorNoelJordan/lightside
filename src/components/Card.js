import React from 'react';
import { Link } from 'react-router-dom';

// import faveInactive from '../images/002-prize.svg';
import './Card.scss'

const Card = ({ data, type }) => {
console.log(data)

    return (
        <article className='card'>
            <h2>Name: {data.name}</h2>
            <h3>{
                (data.birth_year && `Birth Year: ${data.birth_year}` )
                || (data.terrain && `Terrain: ${data.terrain}`) 
                || (data.model && `Model: ${data.model}`)
                }</h3>
            <h3>{
                (data.gender && `Gender: ${data.gender}` )
                || (data.diameter && `Diameter: ${data.diameter}`)
                || (data.vehicle_class && `Class: ${data.vehicle_class}`)
                }</h3>
            <h3>{ 
                (data.height && `Height: ${data.height}`) 
                || (data.population && `Population: ${data.population}`)
                || (data.passengers && `Passengers: ${data.passengers}`)
                || ''}</h3>
            <h3>{ 
                (data.eye_color && `Eye Color: ${data.eye_color}` )
                || (data.climate && `Climate: ${data.climate}` )
                || ''}</h3>

            <Link key={data.name} to={`/${type}/${data.name}`}>More Details</Link>
        </article>
    )
}

export default Card;