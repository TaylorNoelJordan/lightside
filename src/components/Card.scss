.card,
.card-detail {
    background-color: #263957;
    border-radius: 8px;
    color: white;
    margin: 15px;
    opacity: 0.9;
    padding: 10px;
    // display: inline-block;
}

.card {
    width: 200px;
}

.card:hover {
    box-shadow: 1px 1px 1px 1px  #646370;
    cursor: pointer;
    transform: scale(1.04);
    transition: all ease-in-out .3s;
  }

  .card-detail {
      width: auto; 
  }

.favIcon {
    float: right;
    height: 30px;
    // margin: 15px;
}

.detailsLink,
.back-btn {
    color: #FEF1CD;
    font-family: 'AnakinMono', sans-serif;
    font-size: 14px;
}