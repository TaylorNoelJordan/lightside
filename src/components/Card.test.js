import React from 'react';
import Card from '../components/Card';
import { shallow } from 'enzyme';

describe('Card', () => {
    let wrapper;
    let mockToggleFavorite;
    let mockAddFavorite;
    let mockRemoveFavorite;
    let mockData;
    let mockFavorites;

    beforeEach(() => {
        mockToggleFavorite = jest.fn();
        mockAddFavorite = jest.fn();
        mockRemoveFavorite = jest.fn();
        mockFavorites = [
            {favorite1: 'Luke Skywalker'},
            {favorite2: 'Leia SkyWalker'}
        ]
        mockData = {
            name: 'Beyonce',
            birth_year: 1990,
            gender: 'fierce',
            height: 'tall',
            eye_color: 'golden'
        }
        wrapper = shallow(
            <Card 
                data={mockData}
                id='1'
                key='100'
                toggleFavorite={mockToggleFavorite}
                addFavorite={mockAddFavorite}
                removeFavorite={mockRemoveFavorite}
                favorites={mockFavorites}
            />);
    });

    it('should render all passed data', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('should set state when toggleFavorite is called', () => {
        wrapper.instance().setState({ isFavorited: true });
        wrapper.instance().toggleFavorite();

        expect(wrapper.state('isFavorited')).toEqual(false)

    });

    it('should call addFavorite after setting state', () => {
        wrapper.instance().toggleFavorite();
        wrapper.instance().setState({ isFavorited: true })

        expect(wrapper.instance().props.addFavorite).toHaveBeenCalled();
    });

    it('should call removeFavorite if current favorite exists in Favorites on toggleFavorite', () => {
        wrapper.instance().props.addFavorite(mockData);
        wrapper.instance().toggleFavorite();
 
        expect(wrapper.instance().props.favorites).toEqual(mockFavorites)
    });

})