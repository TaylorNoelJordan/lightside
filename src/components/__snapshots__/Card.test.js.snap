// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render all passed data 1`] = `undefined`;
